### PR TITLE
Fixed folder can be renamed to empty string "" #989

### LINKED ISSUE #989: Folder can be renamed to "" empty string
#### Nature of issue?

- Found a bug

#### Details about the bug: 
It is possible to clear the name of a folder by renaming it to "" and files can be renamed to only contain a valid extension (".js", ".css" etc.).
Moreover two or more folders with an empty name ("") can be contained in the same folder which will probably cause issues.

- Web browser and version: Chrome 71.0.3578.98
- Operating System: Windows and Linux
- Steps to reproduce this bug: 
    1. Create a folder with a proper name.
    2. Click rename in the file navigator.
    3. Clear the name of the folder and click outside of it. 

Screenshot_20190325_220842

### PR DESCRIPTION
**Pull request by olalekanteeblaze**
I have verified that this pull request:

* [x] has no linting errors (`npm run lint`)
* [x] is from a uniquely-named feature branch and has been rebased on top of the latest master. (If I was asked to make more changes, I have made sure to rebase onto master then too)
* [x] is descriptively named and links to an issue number, i.e. `Fixes #989 `

### CODE REVIEW

**Comment by catarak** [comment]
there's a function in `FileNode.jsx` called `validateFileName`, which should be updated, rather than `handleFileNameChange`

**Comment by olalekanteeblaze** [comment]
I've updated it. I think adding a length cap should fix it

File: client/modules/IDE/components/FileNode.jsx [hunk 1]
```diff
@@ -77,6 +77,9 @@ export class FileNode extends React.Component {
     ) {
       this.props.updateFileName(this.props.id, this.originalFileName);
     }
+    if (this.props.name.length) {
```
Comment (catarak) [inline client/modules/IDE/components/FileNode.jsx]: i think this needs to come first in the `if` statements. also i think the logic is messed up here—it should be `if` and `if else` statements here so the file name doesn't get updated multiple times.

**Comment by catarak** [comment]
i just checked out this branch and tested it, and i'm still able to rename a folder to a empty string.

**Pull request closed without merging**